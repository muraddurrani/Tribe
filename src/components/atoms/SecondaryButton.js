import React from 'react'
import { StyleSheet } from 'react-native'
import { Button } from 'react-native-elements'
import theme from '../../styles/theme'

function SecondaryButton(props) {
  return (
    <Button
      buttonStyle = {styles.button}
      titleStyle = {styles.title}
      title = {props.title}
    />
  )
}

const styles = StyleSheet.create({
  button: {
    borderWidth: 1,
    borderColor: theme.colours.gray6,
    backgroundColor: theme.colours.gray0
  },
  title: {
    color: theme.colours.gray6
  }
})

export default SecondaryButton
