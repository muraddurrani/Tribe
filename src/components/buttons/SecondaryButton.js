import React from 'react'
import { StyleSheet } from 'react-native'
import { Button } from 'react-native-elements'
import colours from '../../styles/colours'

function SecondaryButton({containerStyle, buttonStyle, titleStyle, ...props}) {
  return (
    <Button
      {...props}
      containerStyle = {{...styles.container, ...containerStyle}}
      buttonStyle = {{...styles.button, ...buttonStyle}}
      titleStyle = {{...styles.title, ...titleStyle}}
    />
  )
}

const styles = StyleSheet.create({
  container: {
    width: 120,
    height: 40,
    borderRadius: 20
  },
  button: {
    backgroundColor: 'transparent',
    width: 120,
    height: 40,
    borderRadius: 20,
    borderWidth: 2,
    borderColor: colours.midpoint1
  },
  title: {
    color: colours.midpoint1
  }
})

export default SecondaryButton
