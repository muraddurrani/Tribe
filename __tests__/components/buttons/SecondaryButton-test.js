import React from 'react'
import renderer from 'react-test-renderer'
import SecondaryButton from '../../../src/components/buttons/SecondaryButton'

it('SecondaryButton renders correctly', () => {
  const tree = renderer
    .create(<SecondaryButton title = 'Secondary Button'/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})