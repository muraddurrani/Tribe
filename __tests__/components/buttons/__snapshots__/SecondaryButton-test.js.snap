// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 120,
      },
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#FF6C63",
          "borderRadius": 20,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "padding": 8,
          "width": 120,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF6C63",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        Secondary Button
      </Text>
    </View>
  </View>
</View>
`;
